src/dlsnode/storage/util/InputBuffer.d

```diff
@@ -230,6 +348,71 @@ struct InputBuffer
             return already_copied_to_output + to_copy_from_internal;
         }
     }
+
+    /***************************************************************************
+
+        Reorders buffer in the way that all unused elements are moved to the
+        front of the buffer, making the end of the buffer available for a refill.
+
+        Returns:
+            amount of remaining elements in the buffer
+
+    ***************************************************************************/
+
+    private size_t moveRemainingToFront ()
+    {
+        auto remaining = this.remainingInBuffer();
+
+        void* src = buffer[].ptr + this.position_in_chunk;
```

Comment:
`this.buffer`.

How is `buffer[].ptr` different from `buffer.ptr`, by the way? I think the `[]` is redundant.